build/Dockerfile

```diff
@@ -1,7 +1,11 @@
 # This dockerfile is used to build Mattermost rtcd
-# A multi stage build, with golang:1.17.9-alpine used as a builder
+# A multi stage build, with golang:${GO_VERSION}-alpine used as a builder
 # and gcr.io/distroless/static as runner
-FROM golang:1.17.9-alpine@sha256:5c2fcfeb58ad9d4948d94e7b2d0432a9bc38bee0f8dfb41d383f38a18b75c38d as builder
+
+# GO_VERSION holds the go version we will build the image against
+# automatically generated from go.mod file using makefile
+ARG GO_VERSION=1.17
+FROM golang:${GO_VERSION}-alpine as builder
```

Comment:
we need to pin a version and SHA.

Reply:
This will break all functionality with using dynamic values to go version, depending on the go.mod file. 
Thus will revert https://github.com/mattermost/rtcd/pull/36/commits/f9fe58f21820675babfa6f2b14b69541a0a2f98a

Reply:
Would you be ok with commiting this for now, and work on a future second iteration to providing a process to-pre populating SHA hashes for GOLANG images, and using those dynamically?

Reply:
Yeah we discussed on Friday with Agniva and unfortunately we need always to pin to a specific version and SHA and after soak time of patch releases to adopt them.

Reply:
Cool will drop the commit